flytekit/extras/tasks/shell.py

```diff
@@ -157,7 +157,15 @@ def _run_script(script) -> typing.Tuple[int, str, str]:
     :return: tuple containing the process returncode, stdout, and stderr
     :rtype: typing.Tuple[int, str, str]
     """
-    process = subprocess.Popen(script, stdout=subprocess.PIPE, stderr=subprocess.PIPE, bufsize=0, shell=True, text=True)
+    process = subprocess.Popen(
+        script,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        bufsize=0,
+        shell=True,
+        text=True,
+        executable=executable,
```

Comment:
can we rename this to `shell`? Also, update the docstring.

Reply:
done!